### PR TITLE
Fixes all the regal rat issues

### LINKED ISSUE #63758: Rat Kings have no night vision/are unable to activate abilities + more.
Reporting client version: 514.1572

## Round ID:
175807

## Testmerges:
- [Revert "Require Node 16"](https://www.github.com/tgstation/tgstation/pull/63331)
- [makes ai static into images added to obscured turfs to save on maptick](https://www.github.com/tgstation/tgstation/pull/63666)

## The Details:

Spawned as a Rat King through random event. Upon spawning in I had zero night vision and neither of the two action button abilites (Rat King's Domain and Raise Army) worked when pressed, clicking it had no feedback. 
No Night vision:
dreamseeker_phlSDc9SUO

Additionally, the disposals rummaging ability (triggered on interacting with disposals bins) causes the Rat King to attack the bin and start the doafter at the same time. Perhaps move the disposals rummaging to RMB?
When the Rat King find cheese through this method, the Rat King is supposed to be forced into saying a few lines; rather it seems the disposals bin says the message rather than the Rat King:
dreamseeker_yMTUyHjuK2

### PR DESCRIPTION
**Pull request by SuperNovaa41**
## About The Pull Request
A couple refactors left the regalrat in a broken state
This fixes that

## Why It's Good For The Game
Fixes https://github.com/tgstation/tgstation/issues/63758

## Changelog

:cl:
fix: The regal rat has full access to it's abilities again.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
can you also fix ghosts not being able to enter regal rats after they're initially created? some sometimes spawn in maint for seemingly that purpose, but nobody can take them because the poll is made before you're allowed to observe. it's also a massive bummer if you create one as a chef while there are no ghosts.

could probably just copy+paste venus flytrap code

**Comment by SuperNovaa41** [comment]
ok

File: code/modules/mob/living/simple_animal/hostile/regalrat.dm [hunk 1]
```diff
@@ -53,6 +54,32 @@
...
+ * Asks the interacting ghost if they would like to control the rat.
+ * If they answer yes, and another ghost hasn't taken control, sets the ghost to control the rat.
+ * Arguments:
+ * * mob/user - The ghost to possibly control the rat
+ */
+/mob/living/simple_animal/hostile/regalrat/proc/get_clicked_player(mob/user)
+	if(key || stat)
+		return
+	var/rat_ask = tgui_alert(usr, "Become the Royal Rat?", "Are you sure?", list("Yes", "No"))
+	if(rat_ask == "No" || QDELETED(src))
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/regalrat.dm:75]:
```suggestion
	if(rat_ask != "Yes" || QDELETED(src))
```

File: code/modules/mob/living/simple_animal/hostile/regalrat.dm [hunk 2]
```diff
@@ -109,6 +133,10 @@
 	else
 		SEND_SIGNAL(target, COMSIG_RAT_INTERACT, src)
 
+	if (DOING_INTERACTION(src, "regalrat")) // check again in case we started interacting
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/regalrat.dm:136]: Make `"regalrat"` a define.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/regalrat.dm (hunk 1 above)
Comment (SuperNovaa41) [reply code/modules/mob/living/simple_animal/hostile/regalrat.dm:75]: is this really necessary?

Comment (SuperNovaa41) [reply code/modules/mob/living/simple_animal/hostile/regalrat.dm]: oh wait nvm the answer could be null, i see

**Pull request merged**